compiler/cpp/src/thrift/generate/t_py_generator.cc

```diff
@@ -269,7 +275,16 @@ class t_py_generator : public t_generator {
   }
 
   static bool is_immutable(t_type* ttype) {
-    return ttype->annotations_.find("python.immutable") != ttype->annotations_.end();
+    std::map<std::string, std::string>::iterator it = ttype->annotations_.find("python.immutable");
+
+    if (it == ttype->annotations_.end()) {
```

Comment:
I'm confused as to how ttype->annotations_ would get populated with the correct data.  Below in DebugProtoTest.thrift containing MutableException, it says (python.immutable = "false").  Wouldn't that populate ttype->annotations with an entry where the key is "python.immutable" and the value is "false"?  The code here assumes if an entry is found then it is immutable and does not check the value.

Also if someone is going to set an attribute like "python.immutable" to false, that would mean the default without an attribute means it is immutable.  That is what the code does here.  So wouldn't it be better to look for an attribute of "python.mutable" = <any value> and if it is there, it is not immutable?

Reply:
> The code here assumes if an entry is found then it is immutable and does not check the value.

But it does: `else if (it->second == "false")`

> Also if someone is going to set an attribute like "python.immutable" to false, that would mean the default without an attribute means it is immutable.

Only for exceptions.

> So wouldn't it be better to look for an attribute of "python.mutable"

Those annotations are already "boolean" flags, so I see no benefit of adding a reverse spelling as opposed to setting a boolean to `"false"`